cmd/rtrmon/rtrmon.go

```diff
@@ -585,28 +619,55 @@ func NewComparator(c1, c2 *Client) *Comparator {
 	}
 }
 
-func countFirstSeenOnOrBefore(vrps []*VRPJsonSimple, thresholdTimestamp int64) float64 {
+func isDefinitelyVisible(vrp *VRPJsonSimple, thresholdTimestamp int64) bool {
+	return vrp != nil && vrp.Visible && vrp.FirstSeen <= thresholdTimestamp
+}
+
+func isDefinitelyNotVisisble(vrp *VRPJsonSimple, thresholdTimestamp int64) bool {
+	return vrp == nil || (!vrp.Visible && vrp.LastSeen <= thresholdTimestamp)
+}
+
+func countUnmatched(vrps VRPMap, others VRPMap, thresholdTimestamp int64) float64 {
 	count := 0
 
-	for _, vrp := range vrps {
-		if vrp.FirstSeen <= thresholdTimestamp {
-			count++
+	for key, vrp := range vrps {
+		other := others[key]
+		if isDefinitelyVisible(vrp, thresholdTimestamp) {
+			// VRP is definitely visible, count if other side is definitely not visible
+			if isDefinitelyNotVisisble(other, thresholdTimestamp) {
+				count++
+			}
+		} else if isDefinitelyNotVisisble(vrp, thresholdTimestamp) {
+			// VRP is definitely not visible, count if other side is definitely visible
+			if isDefinitelyVisible(other, thresholdTimestamp) {
+				count++
+			}
 		}
 	}
 
 	return float64(count)
 }
 
-func Diff(a, b VRPMap) []*VRPJsonSimple {
-	onlyInA := make([]*VRPJsonSimple, 0)
-	for key, vrp := range a {
-		if _, ok := b[key]; !ok {
-			onlyInA = append(onlyInA, vrp)
+func Diff(a, b VRPMap) VRPMap {
+	onlyInA := make(VRPMap)
+	for key, vrpA := range a {
+		if vrpA.Visible {
+			if vrpB, ok := b[key]; !ok || !vrpB.Visible {
```

Comment:
Account for visibility here, think that is the imortant change.

Reply:
Yes, the maps contain VRPs that are no longer visible and we need to ignore these when calculating the current difference.